### PR TITLE
feat(describe-pp): print reason files with `refmt`

### PR DESCRIPTION
**Pull request by anmonteiro**
- this initial diff has `refmt` hardcoded in `bin/describe_pp.ml` 
- It might be better to add a new field to the dialect and model this use case in the dialect directly too

### CODE REVIEW

File: test/blackbox-tests/test-cases/describe/describe-pp/describe-pp.t/run.t [hunk 1]
```diff
@@ -16,4 +16,6 @@ We also make sure that the dump file is not present
 This also works for reason code
 
   $ dune describe pp src/main_re.re
-  ;;Util.log "Hello, world!"
+  # 1 "_build/default/src/main_re.pp.re.ml"
+  # 1 "src/main_re.pp.re"
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/describe/describe-pp/describe-pp.t/run.t:20]: How come we're printing the location directives now?

Comment (anmonteiro) [reply test/blackbox-tests/test-cases/describe/describe-pp/describe-pp.t/run.t:20]: the tests have a dedicated `refmt` executable that I changed to match the real `refmt` by printing to stdout

**Comment by anmonteiro** [comment]
> * It might be better to add a new field to the dialect and model this use case in the dialect directly too

I changed the initial implementation to this:
- added a field `dump_ast` in dialects
- described those actions for OCaml and Reason
- in `describe pp`, execute the dialect's `dump_ast` if it exists, falling back to OCaml's, known to exist

**Review by rgrinberg — APPROVED** [review]
LGTM. @nojb could you take a look at the dialect changes?

**Comment by anmonteiro** [comment]
I renamed `dump_ast` to `print_ast`, since `dump_ast` was ambiguous (could also be interpreted as "dump AST to a file")

**Comment by anmonteiro** [comment]
perhaps `print_ast` should be versioned?

**Review by nojb — APPROVED** [review]
The dialect changes LGTM. Thanks!

**Pull request merged**